Tests/SonarScanner.MSBuild.PreProcessor.Test/ArgumentProcessorTests.cs

```diff
@@ -217,7 +217,25 @@ public void PreArgProc_Region_Invalid()
         @"The arguments 'sonar.host.url' and 'sonar.scanner.sonarcloudUrl' are both set. Please set only 'sonar.scanner.sonarcloudUrl'.")]
     [DataRow(null, "https://host", null, "https://api", typeof(ServerHostInfo), "https://host", "https://api")]
     [DataRow(null, "https://host", null, null, typeof(ServerHostInfo), "https://host", "https://host/api/v2")]
-    public void PreArgProc_Region_Overrides(string region, string hostOverride, string sonarClourUrlOverride, string apiOverride, Type expectedHostInforType, string expectedHostUri, string expectedApiUri, params string[] expectedWarnings)
+
+    [DataRow(null, "https://sonarqube.us", null, null, typeof(CloudHostInfo), "https://sonarqube.us", "https://api.sonarqube.us")]
+    [DataRow(null, "https://SONARQUBE.us/", null, null, typeof(CloudHostInfo), "https://SONARQUBE.us/", "https://api.sonarqube.us")]
+    [DataRow(null, null, "https://sonarqube.us", null, typeof(CloudHostInfo), "https://sonarqube.us", "https://api.sonarqube.us")]
+    [DataRow(null, null, "https://SONARQUBE.us/", null, typeof(CloudHostInfo), "https://SONARQUBE.us/", "https://api.sonarqube.us")]
+    [DataRow(null, "https://sonarqube.us", null, "https://api", typeof(CloudHostInfo), "https://sonarqube.us", "https://api")]
+    [DataRow(null, null, "https://sonarqube.us", "https://api", typeof(CloudHostInfo), "https://sonarqube.us", "https://api")]
+    [DataRow("us", "https://sonarqube.us", null, null, typeof(CloudHostInfo), "https://sonarqube.us", "https://api.sonarqube.us",
+        @"The sonar.region parameter is set to ""us"". The setting will be overriden by one or more of the properties sonar.host.url, sonar.scanner.sonarcloudUrl, or sonar.scanner.apiBaseUrl.")]
+    [DataRow("US", "https://SONARQUBE.us/", null, null, typeof(CloudHostInfo), "https://SONARQUBE.us/", "https://api.sonarqube.us",
+        @"The sonar.region parameter is set to ""US"". The setting will be overriden by one or more of the properties sonar.host.url, sonar.scanner.sonarcloudUrl, or sonar.scanner.apiBaseUrl.")]
+
+    [DataRow(null, "https://sonarcloud.io", null, null, typeof(CloudHostInfo), "https://sonarcloud.io", "https://api.sonarcloud.io")]
```

Comment:
Let's document what happens if you miss-match EU host with EU region, as we do have a logic for that.
```
    [DataRow("us", "https://sonarcloud.io", null, null, typeof(CloudHostInfo), "https://sonarcloud.io", "https://api.sonarcloud.io")]
    [DataRow("us", null, "https://sonarcloud.io", null, typeof(CloudHostInfo), "https://sonarcloud.io", "https://api.sonarcloud.io")]

```

Reply:
I added a "region must be empty" guard before the "region from url" detection:
```cs
var regionFromUrl = region == string.Empty ? KnownRegions.SingleOrDefault(x => UrlEquals(x.ServerUrl, sonarHostUrl) || UrlEquals(x.ServerUrl, sonarCloudUrl)) : null;
```
Because of that we get:
```
DEBUG: Server Url: https://sonarcloud.io
DEBUG: Api Url: https://api.sonarqube.us
Region: us
```

I can remove the guard and we will get your expected result. In both cases we will also get
```
WARNING: The sonar.region parameter is set to "us". The setting will be overriden by one or more of the properties sonar.host.url, sonar.scanner.sonarcloudUrl, or sonar.scanner.apiBaseUrl.
```

Reply:
Ok, let's remove the guard. URL will take precedence, and that will also comply with spec of "URLs take precedence" in a way that works. The guard breaks it now

The warning is nice.